File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.swift
let n = Int(readLine()!)!
var d = Array(repeating: 0, count: n+1)

if(n == 1) { print(1) }
else if(n == 2) { print(2) }
else {
    d[1] = 1
    d[2] = 2
    
    for i in 3...n { d[i] = (d[i-1] + d[i-2])%10007 } 
    print(d[n])
}

File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] D = new int[100001];
        
        D[1] = 1; // 2 * 1 크기의 직사각형을 채우기 위해, 2*1 크기 타일 1개 필요
        D[2] = 2; // 2*1 두 개, 1*2 두 개
        
        for(int i=3; i<=n; i++) D[i] = (D[i-1] + D[i-2])%10007;
        System.out.println(D[n]);
    }
}